contracts/src/L1/rollup/L1MessageQueue.sol

```diff
@@ -20,34 +20,25 @@ import {AddressAliasHelper} from "../../libraries/common/AddressAliasHelper.sol"
 contract L1MessageQueue is OwnableUpgradeable, IL1MessageQueue {
     using BitMapsUpgradeable for BitMapsUpgradeable.BitMap;
 
-    /**********
-     * Events *
-     **********/
-
-    /// @notice Emitted when owner updates gas oracle contract.
-    /// @param _oldGasOracle The address of old gas oracle contract.
-    /// @param _newGasOracle The address of new gas oracle contract.
-    event UpdateGasOracle(address indexed _oldGasOracle, address indexed _newGasOracle);
-
-    /// @notice Emitted when owner updates EnforcedTxGateway contract.
-    /// @param _oldGateway The address of old EnforcedTxGateway contract.
-    /// @param _newGateway The address of new EnforcedTxGateway contract.
-    event UpdateEnforcedTxGateway(address indexed _oldGateway, address indexed _newGateway);
-
-    /// @notice Emitted when owner updates max gas limit.
-    /// @param _oldMaxGasLimit The old max gas limit.
-    /// @param _newMaxGasLimit The new max gas limit.
-    event UpdateMaxGasLimit(uint256 _oldMaxGasLimit, uint256 _newMaxGasLimit);
-
     /*************
-     * Variables *
+     * Constants *
      *************/
 
     /// @notice The address of L1ScrollMessenger contract.
-    address public messenger;
+    address public immutable messenger;
 
     /// @notice The address of ScrollChain contract.
-    address public scrollChain;
+    address public immutable scrollChain;
+
+    /*************
+     * Variables *
+     *************/
+
+    /// @dev The storage slot used as L1ScrollMessenger contract, which is deprecated now.
+    address private __messenger;
+
+    /// @dev The storage slot used as ScrollChain contract, which is deprecated now.
+    address private __scrollChain;
 
     /// @notice The address EnforcedTxGateway contract.
     address public enforcedTxGateway;
```

Comment:
Should we make `enforcedTxGateway` and `gasOracle` immutable too?

Reply:
`gasOracle` maybe changed often, also we won't use `L1MessageQueue` as the implementation of `L1MessageQueue` proxy.

`enforcedTxGateway` can be changed to immutable.